server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java

```diff
@@ -765,36 +769,27 @@ public Fuzzy(StreamInput in) throws IOException {
 
         @Override
         public IntervalsSource getSource(SearchExecutionContext context, MappedFieldType fieldType) {
-            NamedAnalyzer analyzer = fieldType.getTextSearchInfo().getSearchAnalyzer();
+            NamedAnalyzer analyzer = null;
             if (this.analyzer != null) {
                 analyzer = context.getIndexAnalyzers().get(this.analyzer);
             }
-            IntervalsSource source;
             if (useField != null) {
                 fieldType = context.getFieldType(useField);
                 assert fieldType != null;
-                checkPositions(fieldType);
-                if (this.analyzer == null) {
-                    analyzer = fieldType.getTextSearchInfo().getSearchAnalyzer();
-                }
             }
-            checkPositions(fieldType);
-            BytesRef normalizedTerm = analyzer.normalize(fieldType.name(), term);
-            FuzzyQuery fq = new FuzzyQuery(new Term(fieldType.name(), normalizedTerm),
-                fuzziness.asDistance(term), prefixLength, 128, transpositions);
-            source = Intervals.multiterm(fq.getAutomata(), term);
+            if (analyzer == null) {
+                analyzer = fieldType.getTextSearchInfo().getSearchAnalyzer();
+            }
+            // Fuzzy queries only work with unicode content so it's legal to call utf8ToString here.
+            String normalizedTerm = analyzer.normalize(fieldType.name(), term).utf8ToString();
+            IntervalsSource source = fieldType.fuzzyIntervals(normalizedTerm, fuzziness.asDistance(term),
+                prefixLength, transpositions, context);
             if (useField != null) {
                 source = Intervals.fixField(useField, source);
             }
             return source;
         }
 
-        private void checkPositions(MappedFieldType type) {
```

Comment:
Should we change the check to still throw a comprehensive exception when positions are not available on the field ?

Reply:
The check has moved to `TextFieldType` so that things would still work with `MatchOnlyTextFieldMapper` which doesn't index positions but computes them on the fly. Does it address your question?

Reply:
ah right I skipped that part. Thanks, that addresses my question.